docs/quickstarts/astro.mdx

```diff
@@ -0,0 +1,211 @@
+---
+title: Astro Quickstart
+description: Add authentication and user management to your Astro app with Clerk.
+---
+
+# Astro Quickstart
+
+<TutorialHero
+  framework="astro"
+  beforeYouStart={[
+    {
+      title: "Set up a Clerk application",
+      link: "/docs/quickstarts/setup-clerk",
+      icon: "clerk",
+    },
+    {
+      title: "Create a Astro application",
+      link: "https://docs.astro.build/en/install-and-setup",
+      icon: "astro",
+    },
+  ]}
+  exampleRepo={[
+    {
+      title: "Astro Quickstart Repo",
+      link: "https://github.com/clerk/clerk-astro-quickstart"
+    }
+  ]}
+>
+
+- Install `@clerk/astro`
+- Set up your environment keys to test your app locally
+- Add TypeScript declarations
+- Add the `clerk()` integration to your application
+- Use Clerk middleware to implement route-specific authentication
+- Create a header with Clerk components for users to sign in and out
+
+</TutorialHero>
+
+<Steps>
+### Install `@clerk/astro`
+
+Clerk's [Astro SDK](/docs/references/astro/overview) has prebuilt components and helpers to make user authentication easier.
+
+To get started using Clerk with Astro, add the SDK to your project:
+
+<CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
+  ```bash filename="terminal"
+  npm install @clerk/astro
+  ```
+
+  ```bash filename="terminal"
+  yarn add @clerk/astro
+  ```
+
+  ```bash filename="terminal"
+  pnpm add @clerk/astro
+  ```
+</CodeBlockTabs>
+
+### Set environment variables
+
+<SignedIn>
+
+Add the following keys to your `.env.local` file. These keys can always be retrieved from the [API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page of your Clerk Dashboard.
+
+</SignedIn>
+
+<SignedOut>
+
+1. Navigate to the Clerk Dashboard.
+2. In the navigation sidebar, select [API Keys](https://dashboard.clerk.com/last-active?path=api-keys).
+3. In the **Quick Copy** section, copy your Clerk publishable and secret key.
+4. Paste your keys into your `.env.local` file.
+
+The final result should resemble the following:
+
+</SignedOut>
+
+  <InjectKeys>
+
+  ```sh filename=".env"
+  PUBLIC_CLERK_PUBLISHABLE_KEY={{pub_key}}
+  CLERK_SECRET_KEY={{secret}}
+  ```
+
+  </InjectKeys>
+
+
+### Add TypeScript declarations
+
+Inside your `src/` directory, update the `.env.d.ts` file to ensure that the locals added by the middleware are properly typed.
+
+```ts filename="src/.env.d.ts" {3}
+/// <reference types="astro/client" />
+/// <reference types="@clerk/astro/env" />
+```
+
+### Update `astro.config.mjs`
+
+To configure Clerk in your astro application, you will need to update your `astro.config.mjs`.
+
+- Add the Clerk integration to the `integrations` list.
+- Install an [SSR adapter](https://docs.astro.build/en/guides/server-side-rendering/#official-adapters). This is required for the integration to work. For this quickstart we chose the [`@astrojs/node`](https://docs.astro.build/en/guides/integrations-guide/node/) adapter.
```

Comment:
Hey Kev, the reason we added this was to ensure that people who do `build and preview` will have a smooth experience testing their app in a local prod setup.

Reply:
I'm not entirely opposite to this suggestion, and I might revisit this one, even if we merge it as is for now.

Reply:
I agree, but this might be confusing because it implies it only works with the Node adapter, while it works with any adapter that is based on Node, such as Netlify or Vercel. Also when running the build command, if the adapter is not set and the output is server it lets you know that you need an adapter to be able to build. Similar to running preview mode without running a build before.

Reply:
Let's just update the comment then to something like:
```suggestion
- Add the Clerk integration to the `integrations` list.
- Install an [SSR adapter](https://docs.astro.build/en/guides/server-side-rendering/#official-adapters). This is required for the integration to work. For this quickstart we chose the [`@astrojs/node`](https://docs.astro.build/en/guides/integrations-guide/node/) adapter. You can use any Node based adapter you wish.
```